.gitlab/e2e.yml

```diff
@@ -69,16 +70,18 @@ k8s-e2e-otlp-dev:
 k8s-e2e-otlp-main:
   extends: .k8s_e2e_template
   rules: !reference [.on_main]
-  # needs:
-  #   - dev_master-a6
-  #   - dev_master-a7
   script:
     - inv -e e2e-tests --agent-image=datadog/agent-dev:master-py3 --dca-image=datadog/cluster-agent-dev:master --argo-workflow=otlp
 
 .new_e2e_template:
   stage: e2e
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/test-infra-definitions/runner$TEST_INFRA_DEFINITIONS_BUILDIMAGES_SUFFIX:$TEST_INFRA_DEFINITIONS_BUILDIMAGES
   tags: ["arch:amd64"]
+  needs:
+    - job: "deploy_rpm_testing-a6_x64" # This deploy is the last one of all the deploys
+      optional: true
+    - job: "deploy_rpm_testing-a7_x64" # In case agent6 is not built
+      optional: true
```

Comment:
As per PR description, this is maybe fragile, we might want to put all the jobs of the `kitchen_deploy` stage. Jobs having a needs currently depended only on `deploy_windows` and `deploy_deb_a7_x64`

Reply:
Yes, this is very fragile, it relies on the assumption that RPM builds and deploys are always slower than the rest, which doesn't hold in the general case.

IMO, we could add all `kitchen_deploy` jobs here as a stopgap, but in practice, individual jobs should know what they need, and describe their dependencies accordingly.

Reply:
If the jobs with dependencies explicitly specified currently depend only on `deploy_windows` and `deploy_deb_a7_x64`, why wouldn't we just specify those same dependencies here? For reference our `new-e2e-process-dev` job uses those same dependencies: https://github.com/DataDog/datadog-agent/blob/9b07d3e8bd6fe91f2fac529db7c954fa1dc6580a/.gitlab/e2e.yml#L281

Reply:
> Member

I was proposing to change this because all the `new-e2e` tests here depend on the 2 same jobs from `kitchen_deploy`, which looks like a bad copy/paste. As Kylian said, we would probably prefer something more explicit, each test should depend on some specific package and/or image build.

Reply:
For the Windows dependencies at least, they were explicitly added to address flakes in Windows tests: https://github.com/DataDog/datadog-agent/pull/21863